### ISSUE TITLE
Migrating to v1.9.1 from v1.8.1 with missing manifests prevents startup

### ISSUE THREAD

**Issue by mxplusb** [goharbor/harbor#9691]
**Expected behavior and actual behavior:**

After migrating from v1.8.1 to v1.9.1, I found I was unable to log in with any user. Debugging proved migrating from v1.8.1 to v1.9.1 with images missing manifests prevents the quota sync from starting properly.

You will see log messages similar to this:

```
Oct 31 14:10:44 172.18.0.1 core[1106]: 2019-10-31T14:10:44Z [ERROR] [/core/api/quota/registry/registry.go:388]: http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/ubuntu","Revision":"sha256:f2557f94cac1cc4509d0483cb6e302da841ecd6f82eb2e91dc7ba6cfd0c580ab"}}]}
Oct 31 14:10:44 172.18.0.1 core[1106]: 2019-10-31T14:10:44Z [ERROR] [/core/api/quota/registry/registry.go:388]: http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/golang","Revision":"sha256:36712b65e4a64f25c8a1a00e0a07ab39922a5b0c1bb9d3b22c0f842c428a89f4"}}]}
Oct 31 14:10:44 172.18.0.1 core[1106]: 2019-10-31T14:10:44Z [INFO] [/core/api/quota/migrator.go:101]: [Quota-Sync]:: fail to dump data from server ... [registry], quit sync ...
Oct 31 14:10:44 172.18.0.1 core[1106]: 2019-10-31T14:10:44Z [ERROR] [/core/main.go:104]: Fail to sync quota data, quota sync error on getting info of project: quota sync error on getting info of repo: http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/ubuntu","Revision":"sha256:f2557f94cac1cc4509d0483cb6e302da841ecd6f82eb2e91dc7ba6cfd0c580ab"}}]}
Oct 31 14:10:44 172.18.0.1 core[1106]: : http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/golang","Revision":"sha256:36712b65e4a64f25c8a1a00e0a07ab39922a5b0c1bb9d3b22c0f842c428a89f4"}}]}Oct 31 14:10:44 172.18.0.1 core[1106]: 2019-10-31T14:10:44Z [FATAL] [/core/main.go:252]: quota migration error, quota sync error on getting info of project: quota sync error on getting info of repo: http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/ubuntu","Revision":"sha256:f2557f94cac1cc4509d0483cb6e302da841ecd6f82eb2e91dc7ba6cfd0c580ab"}}]}
Oct 31 14:10:44 172.18.0.1 core[1106]: : http error: code 404, message {"errors":[{"code":"MANIFEST_UNKNOWN","message":"manifest unknown","detail":{"Name":"library/golang","Revision":"sha256:36712b65e4a64f25c8a1a00e0a07ab39922a5b0c1bb9d3b22c0f842c428a89f4"}}]}
```

The expected behaviour is for those images to be either removed or ignored on migration/startup. 

**Steps to reproduce the problem:**

I think this is how I got into this situation.

1. Run v1.7.4 in pull-through mode.
1. `docker pull` some Docker Hub image, I used `golang` and `ubuntu`.
1. Stop v1.7.4, change to standard/hosting/non-pull through
1. Delete the pull-through images.
1. Follow migration documentation from v1.7.4 to v1.8.1
1. Push some arbitrary images.
1. Follow migration guide from v1.8.1 to v1.9.1

**Versions:**
Please specify the versions of following systems.

- harbor version: v1.9.1
- docker engine version: 19.03.4
- docker-compose version: version 1.23.2, build 1110ad01

**Additional context:**

- **Harbor config files:** 

```yaml
hostname: cr.domain.io
http:
  port: 80
https:
  port: 443
  certificate: /data/cert/domain.org.crt
  private_key: /data/cert/domain.org.key
harbor_admin_password: <admin_password>
database:
  password: <password>
  max_idle_conns: 50
  max_open_conns: 100

data_volume: /data
storage_service:
truststore
  ca_bundle:
clair:
  updaters_interval: 12
jobservice:
  max_job_workers: 10
notification:
  webhook_job_max_retry: 10
chart:
  absolute_url: disabled
log:
  level: info
  local:
    rotate_count: 50
    rotate_size: 200M
    location: /var/log/harbor
_version: 1.9.0
proxy:
  http_proxy:
  https_proxy:
  no_proxy: 127.0.0.1,localhost,.local,.internal,log,db,redis,nginx,core,portal,postgresql,jobservice,registry,registryctl,clair    components:
    - core
    - jobservice
    - clair
```

- **Log files:**

I can send these via [keybase](https://keybase.io/mlloyd) if you want them.

- **How I fixed it:**

1. `docker exec -ti <postgres-cid> bash`
1. `psql -h localhost -p 5432`
1. `\c registry`
1. `delete from repository where name like '%library/golang%';`
1. `delete from repository where name like '%library/ubuntu%';`
1. `\q`
1. `sudo rm -rf /data/registry/docker/registry/v2/blobs/sha256/f2/f2557f94cac1cc4509d0483cb6e302da841ecd6f82eb2e91dc7ba6cfd0c580ab/`
1. `sudo rm -rf /data/registry/docker/registry/v2/blobs/sha256/36/36712b65e4a64f25c8a1a00e0a07ab39922a5b0c1bb9d3b22c0f842c428a89f4/`
1. `sudo rm -rf /data/registry/docker/registry/v2/repositories/library/{golang,ubuntu}`.
1. `sudo docker-compose down`
1. `sudo docker-compose up -d`
1. Attempted login, works just fine.

I think how I hit this bug is very obtuse, but the problem is specific to the quota sync not passing on missing images should likely be addressed.

**Comment by wy65701436** [comment]
It will be fixed in v1.9.2

**Comment by mxplusb** [comment]
Oh, great! I was able to work through the issue and resolve it independently, so that's good to hear a fix is nearly released.

**Comment by wy65701436** [comment]
close it as the fix is in v1.9.2-rc1.
